src/sonic-frr/patch/0006-zebra-kernel-level-graceful-restart.patch

```diff
@@ -0,0 +1,228 @@
+    immediately.
+ 
+-.. option:: -k, --keep_kernel
++.. option:: -K TIME, --graceful_restart TIME
+ 
+-   When zebra starts up, don't delete old self inserted routes.
++   If this option is specified, the graceful restart time is TIME seconds.
++   Zebra, when started, will read in routes.  Those routes that Zebra
++   identifies that it was the originator of will be swept in TIME seconds.
++   If no time is specified then we will sweep those routes immediately.
+ 
+ .. option:: -r, --retain
+ 
+diff --git a/zebra/main.c b/zebra/main.c
+index b54c36c..15145c2 100644
+--- a/zebra/main.c
++++ b/zebra/main.c
+@@ -80,8 +80,7 @@ int retain_mode = 0;
+ /* Allow non-quagga entities to delete quagga routes */
+ int allow_delete = 0;
+ 
+-/* Don't delete kernel route. */
+-int keep_kernel_mode = 0;
++int graceful_restart;
+ 
+ bool v6_rr_semantics = false;
+ 
+@@ -95,12 +94,12 @@ uint32_t nl_rcvbufsize = 4194304;
+ struct option longopts[] = {
+ 	{"batch", no_argument, NULL, 'b'},
+ 	{"allow_delete", no_argument, NULL, 'a'},
+-	{"keep_kernel", no_argument, NULL, 'k'},
+ 	{"socket", required_argument, NULL, 'z'},
+ 	{"ecmp", required_argument, NULL, 'e'},
+ 	{"label_socket", no_argument, NULL, 'l'},
+ 	{"retain", no_argument, NULL, 'r'},
+ 	{"vrfdefaultname", required_argument, NULL, 'o'},
++	{"graceful_restart", required_argument, NULL, 'K'},
+ #ifdef HAVE_NETLINK
+ 	{"vrfwnetns", no_argument, NULL, 'n'},
+ 	{"nl-bufsize", required_argument, NULL, 's'},
+@@ -271,13 +270,14 @@ int main(int argc, char **argv)
+ 	char *netlink_fuzzing = NULL;
+ #endif /* HANDLE_NETLINK_FUZZING */
+ 
++	graceful_restart = 0;
+ 	vrf_configure_backend(VRF_BACKEND_VRF_LITE);
+ 	logicalrouter_configure_backend(LOGICALROUTER_BACKEND_NETNS);
+ 
+ 	frr_preinit(&zebra_di, argc, argv);
+ 
+ 	frr_opt_add(
+-		"bakz:e:l:o:r"
++		"baz:e:l:o:rK:"
+ #ifdef HAVE_NETLINK
+ 		"s:n"
+ #endif
+@@ -289,24 +289,24 @@ int main(int argc, char **argv)
+ #endif /* HANDLE_NETLINK_FUZZING */
+ 		,
+ 		longopts,
+-		"  -b, --batch           Runs in batch mode\n"
+-		"  -a, --allow_delete    Allow other processes to delete zebra routes\n"
+-		"  -z, --socket          Set path of zebra socket\n"
+-		"  -e, --ecmp            Specify ECMP to use.\n"
+-		"  -l, --label_socket    Socket to external label manager\n"
+-		"  -k, --keep_kernel     Don't delete old routes which were installed by zebra.\n"
+-		"  -r, --retain          When program terminates, retain added route by zebra.\n"
+-		"  -o, --vrfdefaultname  Set default VRF name.\n"
++		"  -b, --batch              Runs in batch mode\n"
++		"  -a, --allow_delete       Allow other processes to delete zebra routes\n"
++		"  -z, --socket             Set path of zebra socket\n"
++		"  -e, --ecmp               Specify ECMP to use.\n"
++		"  -l, --label_socket       Socket to external label manager\n"
++		"  -r, --retain             When program terminates, retain added route by zebra.\n"
++		"  -o, --vrfdefaultname     Set default VRF name.\n"
```

Comment:
Can we keep the lines untouched?

Reply:
> Can we keep the lines untouched?

@paulnice: this is changed already on the FRR master. So we will get it anyways in further releases. Thx.
Refer: https://github.com/FRRouting/frr/pull/4301